### PR TITLE
fix: ID3 does not indicate aac data

### PR DESCRIPTION
**Pull request by brandonocasey**
ID3 tags are not specific to aac data. Almost any audio container format can use it. AAC has a specific syncbyte `0xFFF1` that we need to look for instead.

### CODE REVIEW

File: lib/aac/utils.js [hunk 1]
```diff
@@ -24,13 +24,35 @@ var ADTS_SAMPLING_FREQUENCIES = [
   7350
 ];
 
+var parseId3TagSize = function(header, byteIndex) {
```
Comment (brandonocasey) [inline lib/aac/utils.js:27]: Code is unchanged. I just moved it above `isLikelyAacData` as we use it there now.

File: lib/aac/utils.js [hunk 2]
```diff
@@ -24,13 +24,35 @@ var ADTS_SAMPLING_FREQUENCIES = [
...
+
+// TODO: use vhs-utils
 var isLikelyAacData = function(data) {
+  var offset = 0;
+
   if ((data[0] === 'I'.charCodeAt(0)) &&
       (data[1] === 'D'.charCodeAt(0)) &&
       (data[2] === '3'.charCodeAt(0))) {
-    return true;
+    offset = parseId3TagSize(data, 0);
```
Comment (gesinger) [inline lib/aac/utils.js:49]: Should we check the length here before this call?

File: test/aac-utils.test.js [hunk 3]
```diff
@@ -15,8 +15,9 @@ QUnit.test('correctly determines aac data', function(assert) {
   assert.ok(utils.isLikelyAacData(testSegment), 'test segment is aac');
 
   var id3Offset = utils.parseId3TagSize(testSegment, 0);
-
   assert.ok(utils.isLikelyAacData(testSegment.subarray(id3Offset)), 'test segment is aac without id3');
+  assert.notOk(utils.isLikelyAacData(testSegment.subarray(id3Offset + 25)), 'non aac data not recognized');
+  assert.notOk(utils.isLikelyAacData(testSegment.subarray(0, 5)), 'not enouch aac data is not recognized');
```
Comment (gesinger) [inline test/aac-utils.test.js:20]:
```suggestion
  assert.notOk(utils.isLikelyAacData(testSegment.subarray(0, 5)), 'not enough aac data is not recognized');
```

File: lib/aac/utils.js [hunk 4]
```diff
@@ -24,13 +24,35 @@ var ADTS_SAMPLING_FREQUENCIES = [
...
   }
-  return false;
+
+  return data.length >= offset + 2 &&
+    (data[offset] & 0xFF) === 0xFF &&
+    (data[offset + 1] & 0xE0) === 0xE0 &&
```
Comment (gesinger) [inline lib/aac/utils.js:54]: Since the ADTS header syncword is 0xFFF, should this be 0xF0 instead of 0xE0?

File: lib/aac/utils.js [hunk 5]
```diff
@@ -24,13 +24,35 @@ var ADTS_SAMPLING_FREQUENCIES = [
...
+    (data[offset + 1] & 0x16) === 0x10;
```
Comment (gesinger) [inline lib/aac/utils.js:55]: May be worth a comment above that mentions the syncword that's being checked for and that this comparison verifies the 2 layer bits as always 0.

File: lib/aac/utils.js (hunk 4 above)
Comment (gesinger) [reply lib/aac/utils.js:54]: May be worth a comment.

File: lib/aac/utils.js [hunk 6]
```diff
@@ -51,6 +51,8 @@ var isLikelyAacData = function(data) {
 
   return data.length >= offset + 2 &&
     (data[offset] & 0xFF) === 0xFF &&
+    // verify that the 2 layer bits are 0, aka this
```
Comment (gesinger) [inline lib/aac/utils.js:54]: Should this comment be on the next line?

Comment (brandonocasey) [reply lib/aac/utils.js:54]: done

**Pull request merged**